src/file-events/cpp/win_fsnotifier.cpp

```diff
@@ -259,13 +353,12 @@ void Server::registerPath(const u16string& path) {
 void Server::unregisterPath(const u16string& path) {
     u16string longPath = path;
     convertToLongPathIfNeeded(longPath);
-    reportFinished(longPath);
-}
-
-void Server::reportFinished(const u16string path) {
-    if (watchPoints.erase(path) == 0) {
+    auto it = watchPoints.find(longPath);
+    if (it == watchPoints.end()) {
```

Comment:
We should check if the watch point is finished here if we find one for the path.